docs/hugo/content/architecture-decision-records/ADR-2021-02-Property-Conversions.md

```diff
@@ -0,0 +1,101 @@
+---
+title: '2021-02: Property Conversions'
+date: 2021-02-09
+---
+
+# Property Conversions 
+
+## Context
+
+To facilitate the use of dedicated storage variants for persistence of our custom resources, we need to codegen conversion routines that will copy all properties defined on one version of a resource to another version.
+
+Given the way resources evolve from version to version, these conversions need to handle a wide range of conversions between types that are similar, but not identical.
```

Comment:
very minor: Wording: "these conversions need to handle a wide range of conversions between..." seems phrase strangely?

"these conversion functions need to handle..." or conversion routines or "these conversions need to handle a wide range of transformations..."?

Reply:
Reworded.